### PR TITLE
LibGfx+WindowServer+Base: Add TitleButtonsIconOnly flag for themes and use for Coffee and Cupertino

### PR DESCRIPTION
**Pull request by MacDue**
Before          | After
:-------------------------:|:-------------------------:
Screenshot from 2022-04-23 15-34-49 (2) |Screenshot from 2022-04-23 15-47-42

Wanted to fix a little annoyance with the icons on the Coffee theme looking wrong when the title bar colour changes.

I thought it'd just be a quick change to the .png files, but I ended up needed to add a new `TitleButtonsIconOnly` flag for themes
to disable panting the buttons under the icons.

I've set this flag now in both Coffee and Cupertino (Cupertino had the same issue, but was less noticeable as the buttons were dark and the icons bigger).

### CODE REVIEW

**Comment by linusg** [comment]
Related: #13740

**Comment by MacDue** [comment]
Whoops, didn't notice there's another PR though it's done slightly differently adding `EnableDefaultTitleButtonsBackground` to all themes bar Coffee. 

Should I close this?

File: Userland/Libraries/LibGUI/AbstractThemePreview.cpp [hunk 1]
```diff
@@ -139,7 +139,9 @@ void AbstractThemePreview::paint_window(StringView title, Gfx::IntRect const& re
     painter.fill_rect(rect.translated(-frame_rect.location()), m_preview_palette.color(Gfx::ColorRole::Background));
 
     for (auto& button : buttons) {
-        Gfx::StylePainter::paint_button(painter, button.rect, m_preview_palette, Gfx::ButtonStyle::Normal, false);
+        if (!m_preview_palette.title_buttons_icon_only()) {
+            Gfx::StylePainter::paint_button(painter, button.rect, m_preview_palette, Gfx::ButtonStyle::Normal, false);
+        }
```
Comment (linusg) [inline Userland/Libraries/LibGUI/AbstractThemePreview.cpp:144]: Same here

File: Base/res/themes/Cupertino.ini [hunk 2]
```diff
@@ -75,6 +75,8 @@ TitleAlignment=Center
 
 [Flags]
 IsDark=true
+TitleButtonsIconOnly=true
+
```
Comment (linusg) [inline Base/res/themes/Cupertino.ini:79]: No need for two newlines here

**Review by linusg — CHANGES_REQUESTED** [review]
This one looks nicer, but I'll keep the other one open until it's merged :)

**Pull request merged**